spec.html

```diff
@@ -879,6 +879,12 @@ <h1>Algorithm Conventions</h1>
     <p>Algorithm steps may declare named aliases for any value using the form “Let _x_ be _someValue_”. These aliases are reference-like in that both _x_ and _someValue_ refer to the same underlying data and modifications to either are visible to both. Algorithm steps that want to avoid this reference-like behaviour should explicitly make a copy of the right-hand side: “Let _x_ be a copy of _someValue_” creates a shallow copy of _someValue_.</p>
     <p>Once declared, an alias may be referenced in any subsequent steps and must not be referenced from steps prior to the alias's declaration. Aliases may be modified using the form “Set _x_ to _someOtherValue_”.</p>
 
+    <emu-clause id="sec-parallelism">
+      <h1>Parallelism</h1>
+      <p>A step that ends with “<dfn>in parallel</dfn>” specifies that it is run at the same time as subsequent steps in the algorithm. This specification does not define the precise mechanism by which this is achieved. For example, an implementation may use processes or threads that make use of multiple CPUs or multiple cores of a single CPU. If a parallel step is subdivided, the substeps run sequentially within the unit of parallelism of the parallel step.</p>
```

Comment:
This is the main thing I'd like to discuss in call. cc @tc39/ecma262-editors

Reply:
Probably need to wordsmith agent definitions to allow for parallel steps. In this section I refer to "thread" as the implementation concept to realize parallelism, like HW threads or green threads. In the agent section, I think "execution thread" is a more abstract thing.

The intention is that the agent encompasses a single thread of execution of _observable JS evaluation_. We should probably qualify parallel steps with saying parallel steps are not allowed to affect observable JS evaluation thread. 

Indeed, the parallel step in this proposal shouldn't touch JS values directly. It either wakes up a blocked agent, or it calls out to the host to resolve a promise. In the case of HTML, the host also can't directly resolve promises across agents. What HTML does is it queues up a task on the target agent to perform the resolution. The queuing of the task is the actual parallel operation, the resolution is always done on the agent main execution thread.

Reply:
> "it is run at the same time as subsequent steps in the algorithm" suggests that once those subsequent steps are done (i.e., when the algorithm returns)

No, that's not what I meant. Any suggestions? "At the same time as other steps"? FWIW I don't have the intuition that "subsequent steps" is bounded by the return in the current algorithm.

Reply:
>  FWIW I don't have the intuition that "subsequent steps" is bounded by the return in the current algorithm.

Sure, but "subsequent steps **in the algorithm**" is different. Deleting "in the algorithm" will probably help.

Reply:
Removed parallel steps entirely.